tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,297 @@
+			It("outputs debug logs and the status is accepted and true", func() {
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				ng, err := getNginxGateway(nginxGatewayNsname)
+				Expect(err).ToNot(HaveOccurred())
+
+				gen, err := getNginxGatewayCurrentObservedGeneration(ng)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(verifyNginxGatewayStatus(ng, gen)).To(Succeed())
+
+				Eventually(
+					func() bool {
+						logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+							Container: "nginx-gateway",
+						})
+						if err != nil {
+							return false
+						}
+
+						return strings.Contains(logs, "\"level\":\"debug\"")
+					}).WithTimeout(timeoutConfig.GetTimeout).
+					WithPolling(500 * time.Millisecond).
+					Should(BeTrue())
+			})
+		})
+
+		When("default log level is used", func() {
+			It("only outputs info logs and the status is accepted and true", func() {
+				teardown(releaseName)
+
+				cfg := getDefaultSetupCfg()
+				cfg.debugLogLevel = false
+				setup(cfg)
+
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				ng, err := getNginxGateway(nginxGatewayNsname)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(verifyNginxGatewayStatus(ng, int64(1))).To(Succeed())
+
+				Eventually(
+					func() bool {
+						logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+							Container: "nginx-gateway",
+						})
+						if err != nil {
+							return false
+						}
+
+						return !strings.Contains(logs, "\"level\":\"debug\"")
+					}).WithTimeout(timeoutConfig.GetTimeout).
+					WithPolling(500 * time.Millisecond).
+					Should(BeTrue())
+			})
+		})
+	})
+
+	When("testing on an existing NGF instance", Ordered, func() {
+		BeforeAll(func() {
+			var err error
+			ngfPodName, err = getNGFPodName()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		When("NginxGateway is updated", func() {
+			It("captures the change, the status is accepted and true,"+
+				" and the observed generation is incremented", func() {
+				// previous test has left the log level at info, this test will change the log level to debug
+				ng, err := getNginxGateway(nginxGatewayNsname)
+				Expect(err).ToNot(HaveOccurred())
+
+				gen, err := getNginxGatewayCurrentObservedGeneration(ng)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(verifyNginxGatewayStatus(ng, gen)).To(Succeed())
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).ToNot(ContainSubstring("\"level\":\"debug\""))
+
+				Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+				Eventually(
+					func() bool {
+						ng, err := getNginxGateway(nginxGatewayNsname)
+						if err != nil {
+							return false
+						}
+
+						return verifyNginxGatewayStatus(ng, gen+1) != nil
```

Comment:
This will return true if `verifyNginxGatewayStatus` returns an error

Reply:
Yea this is wrong, I messed this up in the refactor, thank you for catching this.